[release/1.7] Add release notes for v1.7.9

kiashok opened the pull request:
Welcome to the v1.7.9 release of containerd!

The ninth patch release for containerd 1.7 contains various fixes and updates.

### Notable Updates

* **update runc binary to v1.1.10::** ([#9359](https://github.com/containerd/containerd/pull/9359))
* **vendor: upgrade OpenTelemetry to v1.19.0 / v0.45.0** ([#9301](https://github.com/containerd/containerd/pull/9301))
* **Expose usage of cri-api v1alpha2** ([#9336](https://github.com/containerd/containerd/pull/9336))
* **integration: deflake TestIssue9103** ([#9354](https://github.com/containerd/containerd/pull/9354))
* **fix: shimv1 leak issue** ([#9344](https://github.com/containerd/containerd/pull/9344))
* **cri: add deprecation warnings for mirrors, auths, and configs** ([#9327](https://github.com/containerd/containerd/pull/9327))
* **Update hcsshim tag to v0.11.4** ([#9326](https://github.com/containerd/containerd/pull/9326))
* **Expose usage of deprecated features** ([#9315](https://github.com/containerd/containerd/pull/9315))

See the changelog for complete list of changes

Please try out the release binaries and report any issues at
https://github.com/containerd/containerd/issues.

### Contributors

* Samuel Karp
* Kazuyoshi Kato
* Wei Fu
* Kirtana Ashok
* Milas Bowman
* Sebastiaan van Stijn
* ruiwen-zhao

### Changes
<details><summary>26 commits</summary>
<p>

* [release/1.7 backport] update runc binary to v1.1.10 ([#9359](https://github.com/containerd/containerd/pull/9359))
  * [`eff291713`](https://github.com/containerd/containerd/commit/eff29171398e411ab054367f6d9f1892b9d70f67) update runc binary to v1.1.10
* [release/1.7] vendor: upgrade OpenTelemetry to v1.19.0 / v0.45.0 ([#9301](https://github.com/containerd/containerd/pull/9301))
  * [`bd9428ff7`](https://github.com/containerd/containerd/commit/bd9428ff711bda05efecddafe8ca07f568cd994e) vendor: upgrade OpenTelemetry to v1.19.0 / v0.45.0
* [release/1.7] Expose usage of cri-api v1alpha2 ([#9336](https://github.com/containerd/containerd/pull/9336))
  * [`d62cba40c`](https://github.com/containerd/containerd/commit/d62cba40c833f0dacb58084749105fd6cd61283a) Expose usage of cri-api v1alpha2
* [release/1.7] integration: deflake TestIssue9103 ([#9354](https://github.com/containerd/containerd/pull/9354))
  * [`5dbc258a8`](https://github.com/containerd/containerd/commit/5dbc258a81c236040b7ff27382f812c7179d6cd3) integration: deflake TestIssue9103
* [release/1.7] fix: shimv1 leak issue ([#9344](https://github.com/containerd/containerd/pull/9344))
  * [`449912857`](https://github.com/containerd/containerd/commit/449912857d8191c986537af00325d9999922fce3) fix: shimv1 leak issue
* [release/1.7] cri: add deprecation warnings for mirrors, auths, and configs ([#9327](https://github.com/containerd/containerd/pull/9327))
  * [`152c57e91`](https://github.com/containerd/containerd/commit/152c57e918a8374ce2fa20d4eb6ba5896a314529) cri: add deprecation warning for configs
  * [`689a1036d`](https://github.com/containerd/containerd/commit/689a1036dda32e79643b151f632c8da5bac2f149) cri: add deprecation warning for auths
  * [`8c38975bf`](https://github.com/containerd/containerd/commit/8c38975bf22f138f00ddf0fafd4803bbade098a3) cri: add deprecation warning for mirrors
  * [`1fbce40c4`](https://github.com/containerd/containerd/commit/1fbce40c4a7dfad845770580eb146ea2fc64cc46) cri: add ability to emit deprecation warnings
* [release/1.7] Update hcsshim tag to v0.11.4 ([#9326](https://github.com/containerd/containerd/pull/9326))
  * [`73f15bdb6`](https://github.com/containerd/containerd/commit/73f15bdb63e25a13bf99f192f39984e22bc7dbc9) Update hcsshim tag to v0.11.4
* [release/1.7] Expose usage of deprecated features ([#9315](https://github.com/containerd/containerd/pull/9315))
  * [`60d48ffea`](https://github.com/containerd/containerd/commit/60d48ffea657d7d7059dcf497c75d3347e1fd1ab) ctr: new deprecations command
  * [`74a06671a`](https://github.com/containerd/containerd/commit/74a06671ab9396ee94cfee2a588c5d0e170fb698) plugin: record deprecation for dynamic plugins
  * [`fa5f3c91a`](https://github.com/containerd/containerd/commit/fa5f3c91a946795de0173e10a5a3eba4c18aa4c3) server: add ability to record config deprecations
  * [`f7880e7f0`](https://github.com/containerd/containerd/commit/f7880e7f0873024d5307a16d4db8c7a2f360ad49) pull: record deprecation warning for schema 1
  * [`1dd2f2c02`](https://github.com/containerd/containerd/commit/1dd2f2c028bc0bf9b9301a142e09488815c2fb95) introspection: add support for deprecations
  * [`aaf000c18`](https://github.com/containerd/containerd/commit/aaf000c18c047895a891a5dfc7127dc87c034fc9) api/introspection: deprecation warnings in server
  * [`9b7ceee54`](https://github.com/containerd/containerd/commit/9b7ceee540206e6ff6f690676566a73f4f8d443f) warning: new service for deprecations
  * [`b708f8bfa`](https://github.com/containerd/containerd/commit/b708f8bfadcbf95e2acba22ffdeb7d8f8a974151) deprecation: new package for deprecations
</p>
</details>

### Dependency Changes

* **github.com/Microsoft/hcsshim**                                                 v0.11.1 -> v0.11.4
* **github.com/cenkalti/backoff/v4**                                               v4.2.0 -> v4.2.1
* **github.com/go-logr/logr**                                                      v1.2.3 -> v1.2.4
* **github.com/grpc-ecosystem/grpc-gateway/v2**                                    v2.7.0 -> v2.16.0
* **go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc**  v0.40.0 -> v0.45.0
* **go.opentelemetry.io/otel**                                                     v1.14.0 -> v1.19.0
* **go.opentelemetry.io/otel/exporters/otlp/otlptrace**                            v1.14.0 -> v1.19.0
* **go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc**              v1.14.0 -> v1.19.0
* **go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp**              v1.14.0 -> v1.19.0
* **go.opentelemetry.io/otel/metric**                                              v0.37.0 -> v1.19.0
* **go.opentelemetry.io/otel/sdk**                                                 v1.14.0 -> v1.19.0
* **go.opentelemetry.io/otel/trace**                                               v1.14.0 -> v1.19.0
* **go.opentelemetry.io/proto/otlp**                                               v0.19.0 -> v1.0.0

Previous release can be found at [v1.7.8](https://github.com/containerd/containerd/releases/tag/v1.7.8)

kiashok commented:
cc @dmcgowan @dcantah @samuelkarp @mikebrow

samuelkarp commented:
CI is failing:

```
   * 38dd908 "Merge a15f9e33309f6edb73ae9dad2c9e460cba5e6ea9 into ebe25d013d485e849dd3b15ce23c3da547ef4454" ... FAIL
    - FAIL - has whitespace errors. See `git show --check 38dd9085aebd47c0924cab0adbdf5cdc1c31bb8e`.
    - PASS - merge commits do not require DCO
    - PASS - merge commits do not require length check
   * a15f9e3 "Add release notes for v1.7.9" ... FAIL
    - FAIL - has whitespace errors. See `git show --check a15f9e33309f6edb73ae9dad2c9e460cba5e6ea9`.
    - PASS - has a valid DCO
    - PASS - commit subject is 72 characters or less! *yay*
```

Also, the preface appears to be missing from the PR description.  Can you fix the errors and rerun the release tool to generate the PR description?

kiashok commented:
> CI is failing:
> 
> ```
>    * 38dd908 "Merge a15f9e33309f6edb73ae9dad2c9e460cba5e6ea9 into ebe25d013d485e849dd3b15ce23c3da547ef4454" ... FAIL
>     - FAIL - has whitespace errors. See `git show --check 38dd9085aebd47c0924cab0adbdf5cdc1c31bb8e`.
>     - PASS - merge commits do not require DCO
>     - PASS - merge commits do not require length check
>    * a15f9e3 "Add release notes for v1.7.9" ... FAIL
>     - FAIL - has whitespace errors. See `git show --check a15f9e33309f6edb73ae9dad2c9e460cba5e6ea9`.
>     - PASS - has a valid DCO
>     - PASS - commit subject is 72 characters or less! *yay*
> ```
> 
> Also, the preface appears to be missing from the PR description. Can you fix the errors and rerun the release tool to generate the PR description?

let me check. I don't see any whitespace issues when I run git show --check locally

kiashok commented:
@samuelkarp tests are passing now

samuelkarp commented:
The preface appears to be missing from the PR description. Can you rerun the release tool to generate the PR description?

samuelkarp commented on releases/v1.7.9.toml:
```diff
@@ -0,0 +1,33 @@
+
+### Notable Updates
+
+* **cri: add deprecation warnings for mirrors, auths, and configs ([#9327](https://github.com/containerd/containerd/pull/9327))
+* **Update hcsshim tag to v0.11.4 ([#9326](https://github.com/containerd/containerd/pull/9326))
+* **Expose usage of deprecated features ([#9315](https://github.com/containerd/containerd/pull/9315))
```
```suggestion
* **Update hcsshim tag to v0.11.4** ([#9326](https://github.com/containerd/containerd/pull/9326))
* **Expose usage of deprecated features** ([#9315](https://github.com/containerd/containerd/pull/9315))
  * **cri: add deprecation warnings for mirrors, auths, and configs** ([#9327](https://github.com/containerd/containerd/pull/9327))
```

kiashok commented:
> The preface appears to be missing from the PR description. Can you rerun the release tool to generate the PR description?

I am not sure what you mean by preface is missing. I do not see the tool generating anything more than the following:

Welcome to the v1.7.9 release of containerd!

The eighth patch release for containerd 1.7 contains various fixes and updates.

See the changelog for complete list of changes

Please try out the release binaries and report any issues at
https://github.com/containerd/containerd/issues.

### Contributors

* Samuel Karp
* Kirtana Ashok

### Changes
<details><summary>16 commits</summary>
<p>

* [release/1.7] cri: add deprecation warnings for mirrors, auths, and configs ([#9327](https://github.com/containerd/containerd/pull/9327))
  * [`152c57e91`](https://github.com/containerd/containerd/commit/152c57e918a8374ce2fa20d4eb6ba5896a314529) cri: add deprecation warning for configs
  * [`689a1036d`](https://github.com/containerd/containerd/commit/689a1036dda32e79643b151f632c8da5bac2f149) cri: add deprecation warning for auths
  * [`8c38975bf`](https://github.com/containerd/containerd/commit/8c38975bf22f138f00ddf0fafd4803bbade098a3) cri: add deprecation warning for mirrors
  * [`1fbce40c4`](https://github.com/containerd/containerd/commit/1fbce40c4a7dfad845770580eb146ea2fc64cc46) cri: add ability to emit deprecation warnings
* [release/1.7] Update hcsshim tag to v0.11.4 ([#9326](https://github.com/containerd/containerd/pull/9326))
  * [`73f15bdb6`](https://github.com/containerd/containerd/commit/73f15bdb63e25a13bf99f192f39984e22bc7dbc9) Update hcsshim tag to v0.11.4
* [release/1.7] Expose usage of deprecated features ([#9315](https://github.com/containerd/containerd/pull/9315))
  * [`60d48ffea`](https://github.com/containerd/containerd/commit/60d48ffea657d7d7059dcf497c75d3347e1fd1ab) ctr: new deprecations command
  * [`74a06671a`](https://github.com/containerd/containerd/commit/74a06671ab9396ee94cfee2a588c5d0e170fb698) plugin: record deprecation for dynamic plugins
  * [`fa5f3c91a`](https://github.com/containerd/containerd/commit/fa5f3c91a946795de0173e10a5a3eba4c18aa4c3) server: add ability to record config deprecations  * [`f7880e7f0`](https://github.com/containerd/containerd/commit/f7880e7f0873024d5307a16d4db8c7a2f360ad49) pull: record deprecation warning for schema 1
  * [`1dd2f2c02`](https://github.com/containerd/containerd/commit/1dd2f2c028bc0bf9b9301a142e09488815c2fb95) introspection: add support for deprecations
  * [`aaf000c18`](https://github.com/containerd/containerd/commit/aaf000c18c047895a891a5dfc7127dc87c034fc9) api/introspection: deprecation warnings in server  * [`9b7ceee54`](https://github.com/containerd/containerd/commit/9b7ceee540206e6ff6f690676566a73f4f8d443f) warning: new service for deprecations
  * [`b708f8bfa`](https://github.com/containerd/containerd/commit/b708f8bfadcbf95e2acba22ffdeb7d8f8a974151) deprecation: new package for deprecations
</p>
</details>

### Dependency Changes

* **github.com/Microsoft/hcsshim**  v0.11.1 -> v0.11.4

Previous release can be found at [v1.7.8](https://github.com/containerd/containerd/releases/tag/v1.7.8)

samuelkarp commented:
> I am not sure what you mean by preface is missing.

It should include the `preface` defined in `v1.7.9.toml`.  See https://github.com/containerd/containerd/releases for an example of what this should look like, including the "Notable Updates" section.

kiashok commented:
> > I am not sure what you mean by preface is missing.
> 
> It should include the `preface` defined in `v1.7.9.toml`. See https://github.com/containerd/containerd/releases for an example of what this should look like, including the "Notable Updates" section.

Addressed the review comments and pushed new changes.

dmcgowan commented:
Make sure to update https://github.com/containerd/containerd/blob/release/1.7/version/version.go#L26

kiashok commented:
> Make sure to update https://github.com/containerd/containerd/blob/release/1.7/version/version.go#L26

done

estesp commented on releases/v1.7.9.toml:
```diff
@@ -0,0 +1,33 @@
+
+* **Update hcsshim tag to v0.11.4 ([#9326](https://github.com/containerd/containerd/pull/9326))
+* **Expose usage of deprecated features ([#9315](https://github.com/containerd/containerd/pull/9315))
+* **cri: add deprecation warnings for mirrors, auths, and configs ([#9327](https://github.com/containerd/containerd/pull/9327))
```
nit: looks like these 3 notable updates have a start bold (`**`) in markdown but don't end bold on each line?

kiashok replied:
Hi @estesp , I was using the other toml for previous releases as an example before sending out the PR.
![image](https://github.com/containerd/containerd/assets/99994218/9b16780c-d4f0-4ca2-8050-a3d28547108a)


I'll end these lines with bold as you suggested and push a new commit

kiashok replied:
has anything with the release tool changed? I do not see them produce the output as expected

kiashok replied:
I don't remember having to do any of these explicitly in the previous PRs I've sent out to cut new tags

mikebrow reviewed (approved):
LGTM

fuweid commented on releases/v1.7.9.toml:
```diff
@@ -0,0 +1,33 @@
+
+* **Update hcsshim tag to v0.11.4** ([#9326](https://github.com/containerd/containerd/pull/9326))
+* **Expose usage of deprecated features** ([#9315](https://github.com/containerd/containerd/pull/9315))
+* **cri: add deprecation warnings for mirrors, auths, and configs** ([#9327](https://github.com/containerd/containerd/pull/9327))
```
sorry for late reply. would you please add https://github.com/containerd/containerd/pull/9344 in the updates? thanks

kiashok replied:
Adding this now

thaJeztah commented:
Also opened a PR to update the runc binary to v1.1.10;

- https://github.com/containerd/containerd/pull/9359

kiashok commented:
> Also opened a PR to update the runc binary to v1.1.10;
> 
> * [[release/1.7 backport] update runc binary to v1.1.10 #9359](https://github.com/containerd/containerd/pull/9359)

Updated the PR to include this change and @fuweid commit as well. Could you please take a look? Thanks.

fuweid reviewed (approved):
Thanks!

samuelkarp commented:
> Make sure to update [release/1.7/version/version.go#L26](https://github.com/containerd/containerd/blob/release/1.7/version/version.go?rgh-link-date=2023-11-06T23%3A01%3A14Z#L26)

Looks like this is still missing from the PR, can you update it?

dmcgowan commented:
I was expecting to include the Go update in here, not sure if we are still blocked there by the regression.

kiashok commented:
> > Make sure to update [release/1.7/version/version.go#L26](https://github.com/containerd/containerd/blob/release/1.7/version/version.go?rgh-link-date=2023-11-06T23%3A01%3A14Z#L26)
> 
> Looks like this is still missing from the PR, can you update it?

missed the commit during rebase. Brought it back in

kiashok commented:
> I was expecting to include the Go update in here, not sure if we are still blocked there by the regression.

are you talking about this PR - https://github.com/containerd/containerd/pull/9352 ? its still drafted

thaJeztah commented:
- 🙈 I just noticed that https://github.com/containerd/containerd/pull/9301 was merged before the equivalent in main (https://github.com/containerd/containerd/pull/9256), which means that containerd v2.0.0 uses an older version of OTEL than 1.7

thaJeztah commented:
> I was expecting to include the Go update in here, not sure if we are still blocked there by the regression.

My backports are still in review; I asked them for an ETA, but didn't get a response to that (so far)

- https://github.com/golang/go/pull/64091
- https://github.com/golang/go/pull/64092
